modules/pafRtdProvider.js

```diff
@@ -0,0 +1,113 @@
+
+import { submodule } from '../src/hook.js';
+import { mergeDeep, isPlainObject, logMessage, deepSetValue, generateUUID } from '../src/utils.js';
+import { getGlobal } from '../src/prebidGlobal.js';
+import {config} from '../src/config.js';
+
+const SUBMODULE_NAME = 'paf';
+
+/**
+ *
+ * @param {Object} reqBidsConfigObj
+ * @param {function} callback
+ * @param {Object} rtdConfig
+ * @param {Object} userConsent
+ */
+function getBidRequestData(reqBidsConfigObj, onDone, rtdConfig, userConsent) {
+  logMessage('DEBUG(paf):', rtdConfig);
+  let idsAndPreferences;
+  const adUnits = (reqBidsConfigObj.adUnits || getGlobal().adUnits);
+
+  if (rtdConfig.params && rtdConfig.params.proxyHostName && window.PAF) {
+    idsAndPreferences = window.PAF.getIdsAndPreferences();
+    if (!idsAndPreferences) {
+      onDone();
+      return;
+    }
+
+    let transaction_ids = [];
+    for (var i=0; i < adUnits.length; i++) {
+      const uuid = generateUUID();
+      transaction_ids.push(uuid)
+      deepSetValue(adUnits[i], `ortb2Imp.ext.data.paf.transaction_id`, uuid)
+    }
+
+    logMessage('DEBUG(idsAndPreferences):', idsAndPreferences);
+    window.PAF.createSeed({proxyHostName: rtdConfig.params.proxyHostName, callback: function (seed) {setData(seed, rtdConfig, onDone);}}, transaction_ids)
+  } else {
+    onDone();
+    return;
+  }
+}
+
+/**
+ * Lazy merge objects.
+ * @param {Object} target
+ * @param {Object} source
+ */
+ function mergeLazy(target, source) {
+  if (!isPlainObject(target)) {
+    target = {};
+  }
+
+  if (!isPlainObject(source)) {
+    source = {};
+  }
+
+  return mergeDeep(target, source);
+}
+
+function setData(seed, rtdConfig, onDone) {
+  logMessage('DEBUG(seed):', seed);
+  const pafOrtb2 = {
+    ortb2: {
+      user: {
+        ext: {
+          paf: {
+            transmission: {
```

Comment:
According to https://github.com/prebid/addressability-framework/blob/main/mvp-spec/dsp-implementation.md
There should be a "version" property under transmission, it it ok not to set it there ? @OlivierChirouze  @RomainLofaso

Reply:
i omit it because i believe it is set inside the seed, can copy it here if needed